### ISSUE TITLE
[Bug]: Cannot destructure property 'chainId'

### ISSUE THREAD

**Issue by avenbreaks** [MetaMask/metamask-extension#25043]
### Describe the bug

Adding BNB Testnet BEP20 Network, possibly because the network is full, I tried adding the RPC API URL of other platforms such as Alchemy, Chainstack, Publicnode.

### Screenshots/Recordings

![image](https://github.com/MetaMask/metamask-extension/assets/93043826/8ac6c95f-3b2c-4227-8c7f-b9862200e2d4)

![image](https://github.com/MetaMask/metamask-extension/assets/93043826/38d200d0-4d8a-4dd1-b4a5-c934f6ffe8c5)

### Steps to reproduce

1. Add Network
2. Getting Error logs

### Error messages or log output

```shell
Message: Cannot destructure property 'chainId' of '(0 , t.useSelector)(...)' as it is undefined.

Code: TypeError

Stack:
TypeError: Cannot destructure property 'chainId' of '(0 , t.useSelector)(...)' as it is undefined.
  at z (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-2.js:1:841315)
  at ga (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:60103)
  at Os (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:104934)
  at Au (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:90842)
  at Cu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:90767)
  at Tu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:88115)
  at yu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:84859)
  at al (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:112490)
  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:113689
  at xu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:88466)
  at dl (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:113674)
  at render (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-5.js:1:115597)
  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-5.js:1:314517
```
```


### Version

11.16.5

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

**Comment by gauthierpetetin** [comment]
Hi @avenbreaks , thanks for reporting this issue.
Can you please provide more detail regarding steps to reproduce?
When you tried adding RPC API URL of other platforms such as Alchemy, Chainstack, Publicnode, did it fail for every platform or only for some of them?

**Comment by vandan** [comment]
Please specify all the [RPC connection metadat](https://eips.ethereum.org/EIPS/eip-3085)a you included in your `wallet_addEthereumChain` call for "BNB Testnet BEP20 Network"

**Comment by avenbreaks** [comment]
> Hi @avenbreaks , thanks for reporting this issue. Can you please provide more detail regarding steps to reproduce? When you tried adding RPC API URL of other platforms such as Alchemy, Chainstack, Publicnode, did it fail for every platform or only for some of them?

GM, When I add RPC to Metamask and the extension suddenly crashes (cannot be reopened) it brings up a notification like the screenshot above.

for now I really can't open it.

**Comment by SalaGaInCrypto** [comment]
You were able to fix it? Same problem and nothing works.

**Comment by KanteLabs** [comment]
Encountering this issue as well

**Comment by Gudahtt** [comment]
The stack trace points to this line: https://github.com/MetaMask/metamask-extension/blob/c699439fbe523a9214ff58ce7fd681dfc47f9c50/ui/components/app/asset-list/asset-list.js#L58

**Comment by adonesky1** [comment]
Could you link what network you're referring to here:
> BNB Testnet BEP20 Network

Did you mean to refer to a particular asset on BNB Smart Chain Testnet chainId: 97 (0x61)

**Comment by adonesky1** [comment]
Could you link what network you're referring to here:
> BNB Testnet BEP20 Network

Did you mean to refer to a particular asset on BNB Smart Chain Testnet chainId: 97 (0x61)

**Comment by StefanVitor** [comment]
Hi,

I have the same issue, do you have a way to fix this (and that solution to not be to re-install extension)?

**Comment by Gudahtt** [comment]
I suspect that this was fixed by https://github.com/MetaMask/metamask-extension/pull/26327, which was included in v12.0.1

**Comment by Gudahtt** [comment]
Closing as fixed by #26327. No reports so far from v12.0.1.

Everyone in this thread, please update to v12.0.1 and comment here if you notice this problem remains. We can re-open if necessary. If you see a new error, please let us know and open a new issue for it. Thanks!
